packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -123,16 +123,11 @@ const App = (props) => {
     const location = useLocation()
     const authModal = useAuthModal()
     const {site, locale, buildUrl} = useMultiSite()
-
+    const [storeLocatorIsOpen, setStoreLocatorIsOpen] = useState(false)
     const [isOnline, setIsOnline] = useState(true)
     const styles = useStyleConfig('App')
 
     const {isOpen, onOpen, onClose} = useDisclosure()
-    const {
```

Comment:
using conditional rendering instead of `useDisclosure` to ensure that the component gets rendered on click. Otherwise, the component renders on site visit, making unnecessary API calls

Reply:
I realized `useDisclosure` can easily still be used here. So I made the update to put `useDisclosure` back in with this [commit](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1862/commits/3d1eefa1ee3da9a2d65b6a0c4fa0deacc8b163ab)